7.9

ATSchoolENikonov opened the pull request:
первая отправка

methshest committed:
7.9
первая отправка

Separador commented on home_7.9/Task_1/Main.java:
```diff
@@ -0,0 +1,35 @@
+        plants.add(apples);
+        plants.add(tomato);
+        plants.add(banana);
+        plants.stream().filter((p)->
+            p instanceof  Fruit)
+                .forEach(p->{multifruit.add((Fruit)p);
+                    System.out.println(p);});
```
1)  Код плохо отформатирован
2)  В целом вариант `forEach` с лямбдой неплох и работает.
      Будет лучше, если разделить лямбду в `forEach` на отдельные действия:
      -  преобразование в `Plant` во `Fruit`(map)
      -  сбор преобразованных объектов в коллекцию`multifruit`(collect)
      -  печать каждого объекта multifruit (forEach)

Отформатируй этот код и добавь следом для сравнения второй стрим, который  будет добавлять элементы в другой список фруктов по схеме в пункте 2.

methshest committed:
7.9
Вроде переделал, но не уверен, в каком формате должен быть написан первый вариант, должно ли в одну строчку foreach идти или лучше вот так по-строчно

Separador commented on home_7.9/Task_1/Main.java:
```diff
@@ -0,0 +1,43 @@
+//                filter((p) -> p instanceof Fruit).
+//                forEach(p -> {multifruit.add((Fruit) p);
+//                    System.out.println(p);});
+        plants.stream().
+                filter(p -> p instanceof Fruit).
+                map(p -> (Fruit) p).
+                collect(Collectors.toCollection(()->multifruit)).
+                forEach(System.out::println);
```
Точки принято переносить на следующую строчку вместе с последующим вызовом, т.е.:
```java
plants.stream()
      .filter(...)
```
Подробнее можешь посмотреть [в гайдлайне Google](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break).
